src/Framework/AspNetCoreAnalyzers/src/CodeFixes/Dependencies/AddPackageFixer.cs

```diff
@@ -0,0 +1,124 @@
+using Microsoft.CodeAnalysis;
+using Microsoft.CodeAnalysis.CodeActions;
+using Microsoft.CodeAnalysis.CodeFixes;
+using Microsoft.CodeAnalysis.CSharp.Syntax;
+using Microsoft.CodeAnalysis.ExternalAccess.AspNetCore.AddPackage;
+
+namespace Microsoft.AspNetCore.Analyzers.Dependencies;
+
+[ExportCodeFixProvider(LanguageNames.CSharp, Name = nameof(AddPackageFixer)), Shared]
+public class AddPackageFixer : CodeFixProvider
+{
+    public override async Task RegisterCodeFixesAsync(CodeFixContext context)
+    {
+        var root = await context.Document.GetSyntaxRootAsync(context.CancellationToken).ConfigureAwait(false);
+        var semanticModel = await context.Document.GetSemanticModelAsync(context.CancellationToken).ConfigureAwait(false);
+        if (root == null)
+        {
+            return;
+        }
+
+        if (semanticModel == null)
+        {
+            return;
+        }
+
+        var wellKnownTypes = WellKnownTypes.GetOrCreate(semanticModel.Compilation);
+
+        Dictionary<ThisAndExtensionMethod, PackageSourceAndNamespace> _wellKnownExtensionMethodCache = new()
+        {
+            {
+                new(wellKnownTypes.Get(WellKnownTypeData.WellKnownType.Microsoft_Extensions_DependencyInjection_IServiceCollection), "AddOpenApi"),
+                new("Microsoft.AspNetCore.OpenApi", "Microsoft.Extensions.DependencyInjection")
+            },
+            {
+                new(wellKnownTypes.Get(WellKnownTypeData.WellKnownType.Microsoft_AspNetCore_Builder_WebApplication), "MapOpenApi"),
+                new("Microsoft.AspNetCore.OpenApi", "Microsoft.AspNetCore.Builder")
+            }
+        };
+
+        foreach (var diagnostic in context.Diagnostics)
+        {
+            var location = diagnostic.Location.SourceSpan;
+            var node = root.FindNode(location);
+            if (node == null)
+            {
+                return;
+            }
+            var methodName = node is IdentifierNameSyntax identifier ? identifier.Identifier.Text : null;
+            if (methodName == null)
+            {
+                return;
+            }
+
+            if (node.Parent is not MemberAccessExpressionSyntax)
+            {
+                return;
+            }
+
+            var symbol = semanticModel.GetSymbolInfo(((MemberAccessExpressionSyntax)node.Parent).Expression).Symbol;
+            var symbolType = symbol switch
+            {
+                IMethodSymbol methodSymbol => methodSymbol.ReturnType,
+                IPropertySymbol propertySymbol => propertySymbol.Type,
+                ILocalSymbol localSymbol => localSymbol.Type,
+                _ => null
+            };
+
+            if (symbolType == null)
+            {
+                return;
+            }
+
+            var targetThisAndExtensionMethod = new ThisAndExtensionMethod(symbolType, methodName);
+            if (_wellKnownExtensionMethodCache.TryGetValue(targetThisAndExtensionMethod, out var packageSourceAndNamespace))
```

Comment:
Is a dictionary worthwhile for 2 members, or do we anticipate more in the near future?

Reply:
I anticipate that we'll have a larger collection here (see https://github.com/dotnet/aspnetcore/issues/56092). Whether or not `Dictionary` ends up being the right type for the complete set of extension methods remains to be seen but it seemed sufficient for now.